### PR TITLE
[IOTDB-1065] Fix deletion bug while trying to delete data from a timeseries which contains comma

### PR DESCRIPTION
**Pull request by wshao08**
JIRA issue: https://issues.apache.org/jira/projects/IOTDB/issues/IOTDB-1065

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java [hunk 1]
```diff
@@ -127,33 +128,36 @@ private static String encodeDeletion(Deletion del) {
...
+      endTimestamp = Long.parseLong(fields[length - 1]);
+      startTimestamp = Long.parseLong(fields[length - 2]);
+    } catch (NumberFormatException e) {
+      throw new IOException("Invalid timestamp: " + e.getMessage());
+    }
     try {
-      versionNum = Long.parseLong(fields[2]);
+      versionNum = Long.parseLong(fields[length - 3]);
     } catch (NumberFormatException e) {
       throw new IOException("Invalid version number: " + fields[2]);
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java:153]:
```suggestion
      throw new IOException("Invalid version number: " + fields[length - 3]);
```

File: server/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java [hunk 2]
```diff
@@ -127,33 +128,36 @@ private static String encodeDeletion(Deletion del) {
...
+  /**
+   * Decode a deletion record. E.g. "DELETION,root.ln.wf01.wt01.temperature,111,100,300"
+   * the index of field endTimestamp is length - 1, startTimestamp is length - 2,
+   * versionNum is length - 3. Fields in index range [1, length -3) could all belong
+   * to timeseries path in cases that the paths contain comma.
+   */
   private static Deletion decodeDeletion(String[] fields) throws IOException {
-    if (fields.length != 5 && fields.length != 4) {
-      throw new IOException("Incorrect deletion fields number: " + fields.length);
-    }
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java:133]: Consider add this check back?
```
if (fields.length < 4) {
  throw new IOException("Incorrect deletion fields number: " + fields.length);
}
```

File: server/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java (hunk 1 above)
Comment (wshao08) [reply server/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java]: Fixed

File: server/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java (hunk 2 above)
Comment (wshao08) [reply server/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java]: Yes added

**Review by HTHou — APPROVED** [review]

**Review by qiaojialin — COMMENTED** [review]
It's better to upgrade the mod file in this version.

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**